google/cloud/datastore/helpers.py

```diff
@@ -65,16 +64,18 @@ def _get_meaning(value_pb, is_list=False):
         # the rest which may be enum/int values.
         all_meanings = [_get_meaning(sub_value_pb) for sub_value_pb in values]
         unique_meanings = set(all_meanings)
+
         if len(unique_meanings) == 1:
             # If there is a unique meaning, we preserve it.
-            meaning = unique_meanings.pop()
+            return unique_meanings.pop()
         else:  # We know len(value_pb.array_value.values) > 0.
             # If the meaning is not unique, just return all of them.
-            meaning = all_meanings
+            return all_meanings
+
     elif value_pb.meaning:  # Simple field (int32).
-        meaning = value_pb.meaning
+        return value_pb.meaning
 
-    return meaning
+    return None
```

Comment:
To make sure I'm following, do the edits to `_get_meaning()` impact performance, or is this just for readability?

Reply:
Not a big hit:  the three-item `getattr` can be much faster than a `try: ... except AttributeError:`, but likely not the conditional assignment.  OTOH, since in later commits we no longer risk handing the proto-plus wrapper into `_get_meaning`, we could likely pull that `getattr` out.  The other changes to `_get_meaning` are micro-optimizaitons, which neither helped nor hurt on my comparison test.